lib/inspec/schema.rb

```diff
@@ -147,7 +147,7 @@ class Schema
         "license" => { "type" => "string", "optional" => true },
         "summary" => { "type" => "string", "optional" => true },
         "status" => { "type" => "string", "optional" => false },
-        "skip_message" => { "type" => "string", "optional" => true },
```

Comment:
I suspect this should be retained; see other comment.

Reply:
I trust you're right and I'll put it back, but I'd like to know - what's the use case where this breaks for users?

Reply:
(This has nothing to do with the internal RunData API)

This is part of the schema components used to construct the schemae for the JSON output of InSpec - both the JSON profile export and the JSON reporter. We don't know how much the schema are used, but since we are retaining the fields in the output in the JSON reporter when the profile is skipped, if we drop the skip_message field from the schema, the schema will definitely be wrong.

People definitely use the JSON profile export and JSON reporter; we don't know if users are validating against the schema but breaking that would be bad.

Reply:
Thanks very much for explaining - I need to learn more about these bits of InSpec.

I put `skip_message` back in with a comment that `status_message` should now be used for skip reasons.

Reply:
Status_message is a great approach but definitely appreciate Clinton's comments on how this affects downstream users. Would this be a candidate for an InSpec 5 deprecation? (either one to add then or add now for future removal?) 

But before going that far, is there definitely no edge case in which we'd want both a skip_message and a status_message? 🤔  my initial reaction is no; because if something is skipped in theory there ought not to be anymore to say on the subject.

Reply:
I don't think there's an opportunity here to add a deprecation warning since this is just a schema. We can just add it to our list of things to take care of when cutting InSpec 5.0.

I don't foresee a case where 2 statuses are needed. At some point we potentially might need to add code to decide which of 2 statuses trumps the other. But currently once it reaches `failed` or `skipped` state, that should be the profile's final outcome.